components-java-sdk/src/main/java/io/dapr/components/wrappers/StateStoreGrpcComponentWrapper.java

```diff
@@ -0,0 +1,179 @@
+import dapr.proto.components.v1.State.GetRequest;
+import dapr.proto.components.v1.StateStoreGrpc;
+import io.dapr.components.domain.state.BulkGetError;
+import io.dapr.components.domain.state.SetRequest;
+import io.dapr.components.domain.state.StateStore;
+import io.dapr.v1.ComponentProtos;
+import io.dapr.v1.ComponentProtos.FeaturesResponse;
+import io.grpc.stub.StreamObserver;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+/**
+ * A translation layer between a (local) StateStore implementation and Dapr's gRPC StateStore model.
+ */
+public class StateStoreGrpcComponentWrapper extends StateStoreGrpc.StateStoreImplBase {
+  private static final Etag EMPTY_ETAG = Etag.newBuilder().setValue("").build();
+  private static final State.GetResponse EMPTY_GET_RESPONSE = State.GetResponse.newBuilder()
+      .setData(ByteString.EMPTY)
+      .setEtag(EMPTY_ETAG)
+      .build();
+
+  /**
+   * The state store that this component will expose as a service.
+   */
+  private final StateStore stateStore;
+
+  public StateStoreGrpcComponentWrapper(final StateStore stateStore) {
+    this.stateStore = Objects.requireNonNull(stateStore);
+  }
+
+  @Override
+  public void init(final State.InitRequest request, final StreamObserver<State.InitResponse> responseObserver) {
+    Mono.just(request)
+        .flatMap(req -> stateStore.init(req.getMetadata().getPropertiesMap()))
+        // Response is functionally and structurally equivalent to Empty, nothing to fill.
+        .map(response -> State.InitResponse.getDefaultInstance())
+        .subscribe(responseObserver::onNext, responseObserver::onError, responseObserver::onCompleted);
+  }
+
+  @Override
+  public void features(final ComponentProtos.FeaturesRequest request,
+                       final StreamObserver<FeaturesResponse> responseObserver) {
+    Mono.just(request)
+        .flatMap(req -> stateStore.getFeatures())
+        .map(features -> FeaturesResponse.newBuilder()
+            .addAllFeatures(features)
+            .build()
+        )
+        .subscribe(responseObserver::onNext, responseObserver::onError, responseObserver::onCompleted);
+  }
+
+  @Override
+  public void delete(final DeleteRequest request, final StreamObserver<State.DeleteResponse> responseObserver) {
+    Mono.just(request)
+        .map(io.dapr.components.domain.state.DeleteRequest::new) //Convert to local domain
+        .flatMap(stateStore::delete)
+        // Response is functionally and structurally equivalent to Empty, nothing to fill.
+        .map(response -> State.DeleteResponse.getDefaultInstance())
+        .subscribe(responseObserver::onNext, responseObserver::onError, responseObserver::onCompleted);
+  }
+
+  @Override
+  public void bulkDelete(final BulkDeleteRequest request,
+                         final StreamObserver<State.BulkDeleteResponse> responseObserver) {
+    Flux.fromIterable(request.getItemsList())
```

Comment:
Any plans to promote bulk operations to an optional interface for state stores to implement directly?

Reply:
I am doing this in a follow up PR. Just wanted to get these out of the door

Reply:
Just to avoid this getting forgotten: https://github.com/dapr-sandbox/components-java-sdk/issues/15